stellar-core-prometheus-exporter/stellar-core-prometheus-exporter.py

```diff
@@ -38,51 +57,52 @@ def duration_to_seconds(duration, duration_unit):
-        self.info_keys = ['ledger', 'peers', 'protocol_version', 'quorum', 'startedOn', 'state']
-        self.ledger_metrics = {'age': 'age', 'baseFee': 'base_fee', 'baseReserve': 'base_reserve',
-                               'closeTime': 'close_time', 'maxTxSetSize': 'max_tx_set_size',
-                               'num': 'num', 'version': 'version'}
-        self.quorum_metrics = ['agree', 'delayed', 'disagree', 'fail_at', 'missing']
-        # Examples:
-        #   "stellar-core 11.1.0-unstablerc2 (324c1bd61b0e9bada63e0d696d799421b00a7950)"
-        #   "stellar-core 11.1.0 (324c1bd61b0e9bada63e0d696d799421b00a7950)"
-        #   "v11.1.0"
-        self.build_regex = re.compile('(stellar-core|v) ?(\d+)\.(\d+)\.(\d+)(-[^ ]+)?.*$')
+class StellarCoreHandler(BaseHTTPRequestHandler):
+    def log_message(self, format, *args):
+        return
 
     def get_labels(self):
         try:
             response = requests.get(args.info_uri)
             json = response.json()
             build = json['info']['build']
         except Exception:
-            return {}
+            return []
         match = self.build_regex.match(build)
         if not match:
-            return {}
+            return []
 
         if not match.group(5):
             ver_extra = ''  # If regex did not match ver_extra set it to empty string
         else:
             ver_extra = match.group(5).lstrip('-')
 
-        labels = {
-            "ver_major": match.group(2),
-            "ver_minor": match.group(3),
-            "ver_patch": match.group(4),
-            "ver_extra": ver_extra,
-        }
+        labels = [
+            match.group(2),
+            match.group(3),
+            match.group(4),
+            ver_extra,
+        ]
         return labels
 
-    def collect(self):
-        # TODO handle missing labels, probably return 500?
-        labels = self.get_labels()
-        labels_p75 = labels.copy()  # Work on copy of labels variable to avoid other metrics getting quantile label
-        labels_p75.update({'quantile': '0.75'})
-        labels_p99 = labels.copy()  # Work on copy of labels variable to avoid other metrics getting quantile label
-        labels_p99.update({'quantile': '0.99'})
+    def set_vars(self):
+        self.info_keys = ['ledger', 'peers', 'protocol_version', 'quorum', 'startedOn', 'state']
+        self.ledger_metrics = {'age': 'age', 'baseFee': 'base_fee', 'baseReserve': 'base_reserve',
+                               'closeTime': 'close_time', 'maxTxSetSize': 'max_tx_set_size',
+                               'num': 'num', 'version': 'version'}
+        self.quorum_metrics = ['agree', 'delayed', 'disagree', 'fail_at', 'missing']
+        # Examples:
+        #   "stellar-core 11.1.0-unstablerc2 (324c1bd61b0e9bada63e0d696d799421b00a7950)"
+        #   "stellar-core 11.1.0 (324c1bd61b0e9bada63e0d696d799421b00a7950)"
+        #   "v11.1.0"
+        self.build_regex = re.compile('(stellar-core|v) ?(\d+)\.(\d+)\.(\d+)(-[^ ]+)?.*$')
```

Comment:
I think the regex will not match when we use `~` to prefix `ver_extra` instead of a `-`

Reply:
Yeah. I was going to address that in subsequent PRs.
It would be good to merge this one as is because it's a significant redesign but without visible changes so it's easy to verify it works as expected (which I did).
I can then raise more PRs to address other things in the code